ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/LedgerFactory.scala

```diff
@@ -66,3 +74,30 @@ class KeyValueReadWriteFactory(
   }
 
 }
+
+class KeyValueDeduplicationSupportFactory(
+    delegate: ReadWriteServiceFactory,
+    config: Config[_],
+    completionsService: IndexCompletionsService,
+)(implicit materializer: Materializer, ec: ExecutionContext)
+    extends ReadWriteServiceFactory {
+  override def readService(): ReadService = delegate.readService()
+
+  override def writePackagesService(): WritePackagesService = delegate.writePackagesService()
+
+  override def writeService(): WriteService = {
```

Comment:
There were 3 alternatives as to where do this:
- in the Runner, not chosen as to not any more logic in the runner, it should ideally just connect bigger pieces and not be aware of such a low level detail
- as part of the existing ReadWriteServiceFactory, adding the IndexCompletionService as an argument to the `writeService` method - not chosen as this would expose the `IndexCompletionService` in multiple modules where the factory implementation should really not know about it
- as a new decorator for the `ReadWriteServiceFactory` - the implementation, not ideal as it still forces us to wrap the initial factory in this new layer, but feels cleaner than the other options